scripts/run-external.sh

```diff
@@ -1,20 +1,195 @@
+		CLUSTER=$(kubectl config current-context)
+	}
+
+	info "Using cluster - '$CLUSTER'"
+	return 0
+}
+
+run_operator() {
+
+	local api_server ca_file
+
+	local cluster=".clusters[?(@.name == \"$CLUSTER\")].cluster"
+	api_server=$(kubectl config view -o jsonpath="{$cluster.server}")
+	ca_file=$(kubectl config view --raw -o jsonpath="{$cluster.certificate-authority-data}")
+
+	local key_file cert_file
+	local user=".users[?(@.name == \"$CLUSTER\")].user"
+	key_file=$(kubectl config view -o jsonpath="{$user.client-key-data}")
+	cert_file=$(kubectl config view --raw -o jsonpath="{$user.client-certificate-data}")
+
+	info "Running operator against cluster: $CLUSTER"
+	echo "──────────────────────────────────────────────────────────────────"
+	run ./operator \
+		--apiserver="${api_server}" \
+		--ca-file="${ca_file}" \
+		--cert-file="${cert_file}" \
+		--key-file="${key_file}" 2>&1 | tee tmp/operator.log
+}
+
+validate_args() {
+	if [[ -z "$CLUSTER" ]] && ! $USE_DEFAULT_CONTEXT; then
+		err "missing cluster name or --use-default-context"
+		return 1
+	fi
+
+	# ensure both cluster and use-default-context isn't set
+	if ! [[ -z "$CLUSTER" ]] && $USE_DEFAULT_CONTEXT; then
+		err "Cannot provide both cluster name '$CLUSTER' and --use-default-context"
+		return 1
+	fi
+
+	return 0
+}
+
+ensure_operator_not_running() {
+	$SKIP_OPERATOR_RUN_CHECK && {
+		info "skipping operator run check"
+		return 0
+	}
+
+	local po_label='app.kubernetes.io/name=prometheus-operator'
+
+	local po_pods
+	po_pods=$(kubectl get pods -A -l po_label | wc -l)
+
+	[[ "$po_pods" -gt 0 ]] && {
+		warn "Found $po_pods pods matching $po_label in cluster!"
+		echo "If it is safe to continue, rerun the script with --no-operator-run-check option"
+		return 1
+	}
+	ok "No pods with matching label $po_label running in cluster"
+
+	return 0
+}
+
+show_usage() {
+
+	local scr
+	scr="$(basename "$0")"
+
+	read -r -d '' help <<-EOF_HELP || true
+		Usage:
+		────────────────────────────────────────────
+		  ❯ $scr <cluster-name>
+		  ❯ $scr  -h|--help
+
+		Options:
+		⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻
+		  --help | -h                   show this help
+		  --no-operator-run-check | -f: skips the check that ensures no prometheus-operator
+		                                is running in the cluster.
+		  --use-default-context | -c:   Uses current kubernete context to the operator against
+
+		⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻⎻
+		💡 Tips
+		To get all cluster names in current context, run:
+		   ❯ kubectl config get-contexts -o name
+
+		To get current/default cluster names, run:
+		   ❯ kubectl config current-context
+	EOF_HELP
+
+	echo -e "$help"
+	return 0
+}
+
+parse_args() {
+	### while there are args parse them
+	while [[ -n "${1+xxx}" ]]; do
+		case $1 in
+		-h | --help)
+			SHOW_USAGE=true
+			# exit the loop
+			break
+			;;
+		--no-operator-run-check | -f)
+			shift
+			SKIP_OPERATOR_RUN_CHECK=true
+			shift
+			;;
+		--use-default-context | -c)
```

Comment:
@simonpasquier @JoaoBraveCoding this part is new 
./script/run-external.sh -c   # runs against the current context.